src/components/App.js

```diff
@@ -80,6 +89,29 @@ export default class App extends Component {
         }
     }
 
+    getLink = (event) => {
+
+        // This prevents the href="#" on this button from causing a hash change after we set the hash with replaceState below
+        event.preventDefault();
```

Comment:
I've added `evt.preventDefault()` to the `<Button>` component in master, so we can omit this line.